src/main/clojure/clojure/core/matrix.clj

```diff
@@ -1701,6 +1701,119 @@
   ([m]
     (mp/element-max m)))
 
+(defn clamp
+  "Clamps each element in a numerical array between an upper and lower bound.
+  Returns an error if the lower bound is greater that the upper bound.
```

Comment:
I also put that in the spec of the element-clamp protocol:
```
(element-clamp [m a b] 
    "Returns a matrix where the elements are clamped to be within upper and 
    lower bounds specified by a and b. Must throw an exception if a > b.")
```

Should I remove it from there too?